src/main/kotlin/de/bund/digitalservice/useid/events/EventHandler.kt

```diff
@@ -0,0 +1,34 @@
+package de.bund.digitalservice.useid.events
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import java.util.function.Consumer
+
+
+@Service
+class EventHandler {
+    private val log = KotlinLogging.logger {}
+
+    private val consumers: MutableMap<String, Consumer<Event>> = HashMap()
+
+    /**
+     * This method subscribes a consumer to events sent to the given id.
+     */
+    fun subscribeConsumer(id: String, consumer: Consumer<Event>) {
+        consumers[id] = consumer
+        log.info { "New consumer added with id $id, total consumers: ${consumers.size}" }
+    }
+
+    /**
+     * This method publishes the given event to the according consumer. The consumer id is specified in the event.
+     */
+    fun publish(event: Event) {
+        val consumerId = event.consumerId
+        log.info { "Publish event to consumer $consumerId" }
+        consumers[consumerId]!!.accept(event)
```

Comment:
Maybe we should make sure, that the consumerId is available in that map? (In a more.. graceful way?)

Reply:
When the `consumerId` is not available in the map `consumer`, we throw a `NullPointerException` (`!!`), right? I suggested, that we could handle this event maybe more graceful and don't return a general NPE, i.e.

* either handle the exception and return some kind of application-error (`WidgetNotFound` or something), so that the app can show some other error message than if a NPE exception happens at some other code meaning something different
* or prevent that exception by just ignoring the missing value inside the map (so that someone brute-forcing can't get any information at all, but our app won't learn anything either)

Reply:
Good point! I like the idea of ignoring unknown IDs since it keeps it very simple and prevents observing IDs via brute force. If we assume that most of the messages sent to this endpoint are from a valid source (not an attacker) though, we must assume that the messages are required to complete an ongoing flow and ignoring them would mean missing the success information since neither the eID client nor our server will notice (i.e. no retry or error handling possible) which is why I would go for a more descriptive error response.

Reply:
Exactly my train of thought. 🤓 With this, maybe we can handle users closing the browser window or something by displaying a helpful message in the app?!